src/main/cpp/src/shuffle_split.hpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/table/table_view.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <vector>
+
+namespace spark_rapids_jni {
+
+// TODO: this is duplicated from cudf because the cudf function is not marked as constexpr, so it
+// cannot be called on the gpu there is an issue filed against cudf to make
+// is_fixed_point(cudf::data_type type) constexpr. Once that is done, we can remove this
+template <typename T>
+constexpr inline bool is_fixed_point()
+{
+  return std::is_same_v<numeric::decimal32, T> || std::is_same_v<numeric::decimal64, T> ||
+         std::is_same_v<numeric::decimal128, T> ||
+         std::is_same_v<numeric::fixed_point<int32_t, numeric::Radix::BASE_2>, T> ||
+         std::is_same_v<numeric::fixed_point<int64_t, numeric::Radix::BASE_2>, T> ||
+         std::is_same_v<numeric::fixed_point<__int128_t, numeric::Radix::BASE_2>, T>;
+}
+struct is_fixed_point_impl {
+  template <typename T>
+  constexpr bool operator()()
+  {
+    return is_fixed_point<T>();
+  }
+};
+constexpr bool is_fixed_point(cudf::data_type type)
+{
+  return cudf::type_dispatcher(type, is_fixed_point_impl{});
+}
+
+// per-column information, stored in the metadata header
+struct shuffle_split_col_data {
+  cudf::type_id type;
+
+  shuffle_split_col_data(cudf::type_id _type, cudf::size_type _param)
+    : type(_type), _num_children(_param)
```

Comment:
nit: This is a bit confusing and I have to do a double take to know what was happening with the union and all. A comment would be nice.

Reply:
Done.